modcc/module.cpp

```diff
@@ -40,6 +40,15 @@ Module::Module(std::vector<char> const& buffer)
         buffer_.push_back(0);
 }
 
+Module::Module(const char* buffer, size_t count)
+{
```

Comment:
I can see the C++ salt here, but we are scanning twice the `buffer`, first time in `std::distance` and second time in `insert()`. Of course, that won't be a performance issue for the buffer sizes we really get.

Hmm, I could use though @halfflat's sentinel iterators.